src/core/ext/transport/chttp2/transport/http2_settings.cc

```diff
@@ -22,43 +22,133 @@
 
 #include "src/core/ext/transport/chttp2/transport/http2_settings.h"
 
+#include "absl/strings/str_cat.h"
+
+#include "src/core/ext/transport/chttp2/transport/frame.h"
 #include "src/core/lib/gpr/useful.h"
 #include "src/core/lib/transport/http2_errors.h"
 
-const uint16_t grpc_setting_id_to_wire_id[] = {1, 2, 3, 4, 5, 6, 65027, 65028};
+namespace grpc_core {
+
+void Http2Settings::Diff(
```

Comment:
This LGTM generally, but it seems like it would be possible  to now add some unit testing, for initial settings, clamping, etc.?

Reply:
yeah good point... I'd intended to when I wrote this and then got distracted :)

added a handful of tests now